pkg/genericapiserver/api/handlers/rest.go

```diff
@@ -1079,22 +1113,95 @@ func setListSelfLink(obj runtime.Object, req *restful.Request, namer ScopeNamer)
 	return count, err
 }
 
-func getPatchedJS(patchType types.PatchType, originalJS, patchJS []byte, obj runtime.Object) ([]byte, error) {
+func patchObjectJSON(
```

Comment:
Let's get these out of this file.  The time is right - all the helper functions to patch.go.